dataprep/eda/container.py

```diff
@@ -129,7 +129,14 @@ def __init__(self, **param: Any) -> None:
 
         for attr, value in param.items():
             if attr == "layout":
-                setattr(self, "components", components(value))
+                if len(value) == 0:
+                    setattr(self, "components", ("", []))
+                else:
+                    setattr(self, "components", components(value))
+            elif attr == "meta":
+                setattr(self, attr, value)
+                figs = [val for val in value if val != "Stats"]
+                setattr(self, "figs", figs)
             else:
                 setattr(self, attr, value)
 
```

Comment:
Why does the Context class override `__getitem__` and `__getattr__`? I think I must have missed it in the previous PR. I feel there should have a reason here but it would be better to comment it down. Otherwise, this can be removed so that there can be less `setattr` in the rest of the code.

Also I feel it would be better to use if condition here rather than try-catch.

Reply:
When the template engine tries to get the attribute which doesn't exist, the program will throw an error if we don't override `__getitem__` or `__getattr__` because we use the same HTML template to render different components without strict evaluation.

For instance, `plot(df)` and `plot_missing(df, x)` share the same template but what to be rendered is controlled by `Context` object, if the engine reads a non-existing attribute in the object then it will get `None` and renders nothing instead of being interrupted.

Another reason for why it overrides both method is that now getting the value of a key has the same effect as accessing its attribute.

Reply:
I see, can we have that in the code comment?

Reply:
Will do.